lib/param_parsing/terragrunt.go

```diff
@@ -0,0 +1,43 @@
+package param_parsing
+
+import (
+	"fmt"
+	"github.com/hashicorp/hcl2/gohcl"
+	"github.com/hashicorp/hcl2/hclparse"
+	"github.com/warrensbox/terraform-switcher/lib"
+	"log"
+)
+
+const terraGruntFileName = "terragrunt.hcl"
+
+type terragruntVersionConstraints struct {
+	TerraformVersionConstraint string `hcl:"terraform_version_constraint"`
+}
+
+func GetVersionFromTerragrunt(params Params) Params {
+	filePath := params.ChDirPath + "/" + terraGruntFileName
+	if lib.CheckFileExist(filePath) {
+		fmt.Printf("Reading configuration from %s\n", filePath)
+		parser := hclparse.NewParser()
+		hclFile, diagnostics := parser.ParseHCLFile(filePath)
+		if diagnostics.HasErrors() {
+			log.Fatal("Unable to parse HCL file", filePath)
```

Comment:
Would you please sync logging with https://github.com/warrensbox/terraform-switcher/pull/350 ?

Though there's a "hiccup" that I couldn't figure out: https://github.com/warrensbox/terraform-switcher/pull/334#discussion_r1553832532 — `logger.Fatal` doesn't call `os.Exit(1)` for me 😕